security/pkg/server/ca/authenticate/cidr_authenticator.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authenticate
+
+import (
+	"net"
+	"strings"
+
+	"google.golang.org/grpc/peer"
+
+	"istio.io/istio/pilot/pkg/features"
+	"istio.io/istio/pkg/security"
+)
+
+// CidrAuthenticator is delegated authenticator that validates whether
+// caller is in the trusted Cidr range and hence trust the information
+// passed.
+type CidrAuthenticator struct{}
+
+var _ security.DelegatedAuthenticator = &CidrAuthenticator{}
+
+// CanTrustedCaller determines if the caller is in trusted cidr range.
+func (c *CidrAuthenticator) CanTrustCaller(ctx *security.AuthContext) bool {
+	peerInfo, _ := peer.FromContext(ctx.RequestContext)
+	return isTrustedAddress(peerInfo.Addr.String())
+}
+
+func isTrustedAddress(addr string) bool {
+	if len(features.TrustedCIDRRanges) > 0 {
+		cidrs := strings.Split(features.TrustedCIDRRanges, ",")
```

Comment:
So we split, create objects, etc on every single request ? May be ok for an experimental feature, but add at least comments that this can't be the final form. 

Also see my other comments - it would be much easier to review and understand if you kept this code local to XFCC authenticator, without attempting to abstract it too early.

Reply:
Auth only happens on initial connection